ansible/library/lldpctl_facts.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python
+# This ansible module is for gathering lldp facts from SONiC device.
+# It takes two argument 
+# asic_instance_id :- Used to specify LLDP Instance in Multi-asic platforms
+# skip_interface_pattern_list:- Used to specify interface pattern list to be skip for gathering lldp facts.
+import re
+from ansible.module_utils.basic import AnsibleModule
+
+def gather_lldp(module, lldpctl_docker_cmd, skip_interface_pattern_list):
+    rc, output, err = module.run_command(lldpctl_docker_cmd)
+    if output:
+        output_dict = {}
+        current_dict = {}
+        lldp_entries = output.split("\n")
+        skip_interface_pattern_str = '(?:% s)' % '|'.join(skip_interface_pattern_list) if skip_interface_pattern_list else None
+        for entry in lldp_entries:
+            if entry.startswith('lldp'):
+                path, value = entry.strip().split("=", 1)
+                path = path.split(".")
+                if skip_interface_pattern_list and re.match(skip_interface_pattern_str, path[1]):
+                    continue
+                path_components, final = path[:-1], path[-1]
+            else:
+                value = current_dict[final] + '\n' + entry
+
+            current_dict = output_dict
+            for path_component in path_components:
+                current_dict[path_component] = current_dict.get(path_component, {})
+                current_dict = current_dict[path_component]
+            current_dict[final] = value
+        return output_dict
+
+
+def main():
+    module = AnsibleModule(argument_spec=dict(
```

Comment:
What's the motivation for making this an ansible module as opposed to a member function of `duthost` or a library function somewhere in pytest?

Reply:
@daall i didn't have any particular reason. I saw already we have lldcp_facts via SNMP so added accordingly.

Do we have any guidelines when to to have ansible module vs library/function ? May be if the fact is used only by one test case then better to be function than ansible module ?